### DISCUSSION TITLE
Flask, Alembic, PGSQL: Migration hangs after upgrade to 1.13.2 and works ONCE after downgrade to 1.13.0

### DISCUSSION THREAD

**Discussion by hehobid** [sqlalchemy/alembic#1500] [Usage Questions]
**Describe the bug**
We have a comprehensive Flask app with SQLAlchemy and use alembic for schema management.
A couple of days ago (we believe with the arrival of 1.13.2), we are not able to execute any alembic command that touches the database (upgrade, migrate, current) but can execute commands that only touch the migration files (e.g. heads).
When executing a DB command, the flask app starts but nothing happens. PGSQL is new and empty.

If we rebuild the project with alembic==1.13.0, the command execute nicely ONCE. A second time, it just hangs. alembic==1.13.2, rebuild, alembic==1.13.2, rebuild buys us another single run 

**Expected behavior**
We expect 'flask db upgrade' not to silently hang

**To Reproduce**
For us its as described above:
empty setup
docker-compose start
docker exec -ti <image> /bin/bash
flask db upgrade
_works_

then flask -db upgrade (or migrate -m '')
_hangs_

**Error**
No output at all. Even with increased log level
```



**Versions.**
 - OS: Debian 12.2.0-14
 - Python: 3.10
 - Alembic: 3.12.
 - SQLAlchemy:2.0.30
 - Database: 15.4-2.pgdg120+1
 - DBAPI:

**Additional context**

**Have a nice day!**

**Comment by zzzeek** [comment]
it's very unlikely this has anything to do with specific features in alembic.   what you want to do when PostgreSQL hangs is identify what it's hung on , which you do by looking at [pg_stat_activity](https://www.postgresql.org/docs/current/monitoring-stats.html#MONITORING-PG-STAT-ACTIVITY-VIEW) during the hang.  that will provide detail on long running queries as well as deadlocks.

**Reply by zzzeek** [reply]
not sure why this got closed i likely hit the wrong button , sorry
